compiler/transforms/clear_memory_space.py

```diff
@@ -46,3 +47,18 @@ def clear_memory_space(t):
                 )
 
                 op_in_module.function_type = new_function_type
+
+                # change block args ssa values
+                if op_in_module.body.blocks:
+                    old_args = [old_arg for old_arg in op_in_module.body.block._args]
+                    new_args = [
+                        BlockArgument(
+                            clear_memory_space(old_arg.type),
+                            op_in_module.body.block,
+                            index,
+                        )
+                        for index, old_arg in enumerate(old_args)
+                    ]
+                    for old_arg, new_arg in zip(old_args, new_args):
+                        old_arg.replace_by(new_arg)
+                    op_in_module.body.block._args = tuple(new_args)
```

Comment:
This is to change the function body? or ?

Reply:
to change the types of the block arguments of the funciton body (remove memory space).
Because this is the first time there is an unused function argument, this block argument was not converted to not have memory spaces defined, so this had to be resolved